### PR TITLE
Utils for tagging stages

### PR DESCRIPTION
**Pull request by codebien**
This PR implements [the code for tagging stages](https://github.com/grafana/k6/issues/796#issuecomment-1094897787), adding some utils function for getting and tagging with the index (or the computed profile) of the current running stage.

It requires an `xk6` build of https://github.com/grafana/k6/pull/2493 for running the tests.

```shell
$ xk6 build master
$ ./k6 run tests/stages.js
```

### CODE REVIEW

File: src/stages.js
Comment (codebien) [inline src/stages.js]: Opinions?

File: tests/stages.js
Comment (codebien) [inline tests/stages.js]: Apparently, `throw()` doesn't work as expected :thinking:, I'm going to research more in the meanwhile.

File: src/stages.js
Comment (olegbespalov) [inline src/stages.js:5]: I'm wondering, should we maybe also detect somehow negative cases :thinking: 

I mean from what I see (please correct me if I am wrong), but currently:

```js
parseDuration("0") === parseDuration("")
```

File: src/stages.js
Comment (olegbespalov) [inline src/stages.js]:
```suggestion
		throw new Error(`only ramping-vus or ramping-arravial-rate supports stages, it is not possible to get a stage index on other executors.`)
```

File: src/stages.js
Comment (olegbespalov) [reply src/stages.js]: My opinion is that at least `tagWithCurrentStageIndex` and `tagWithCurrentStageProfile` should not use the same key. I mean, what if a customer wants to have them both? :thinking:

File: tests/stages.js
Comment (olegbespalov) [inline tests/stages.js:169]: Maybe some opposing/corner cases?

e.g.:
```
parseDuration("0s")
parseDuration("")
parseDuration("0")
parseDuration("1s 1s 1s")
parseDuration("1w")
```

File: src/stages.js
Comment (olegbespalov) [inline src/stages.js:57]: But will this correctly work with some unexpected units? :thinking: (e.g. `1w`)

**Comment by codebien** [comment]
Added some more edge cases, and applied more refactor for making the code a bit more readable. Let me know if you have better ideas for having a better code for `parseDuration`. It would be good to keep at least O(N) considering the amount of time it could be invoked during a test.

**Review by oleiade — APPROVED** [review]
Nickel Chrome! Great work! 🎉

File: src/stages.js
Comment (MStoykov) [inline src/stages.js:36]: nit: I would recommend moving this whole parse duration in a separate function.

**Review by mstoykov — APPROVED** [review]

**Review by javaducky — APPROVED** [review]

**Comment by codebien** [comment]
Ok, based on the approvals we are ready to open a PR on the docs and the relative PR on `jslib.k6.io`. This PR is dependent on the upcoming `k6 v0.38.0` so I will wait for the new release before merging the PRs.

**Comment by sniku** [comment]
Cool to see "unit tests" using k6chaijs! :tada:

**Pull request merged**
